### PR TITLE
Move Cirq Executors from docs to cirq_utils

### LINKED ISSUE #552: Revamp executor docs
* Add executor examples to the library instead of the docs.
    - This is done for `mitiq_qiskit` in #463.
* Use these docs to instead describe different executor types: single/serial, batched, and raw.

### PR DESCRIPTION
**Pull request by purva-thakre**
Description
-----------
Moves Cirq executors from docs with tests in `tests_cirq_utils`.  `test-cirq` was added in `Makefile`. 
Partially fixes #552

Checklist
-----------
Please make sure you have finished the following tasks before requesting a review of the pull request (PR). For more information, please refer to the [Contributor's Guide](../blob/master/CONTRIBUTING.md):

- [x] Contributions to mitiq should follow the [pep8 style](https://www.python.org/dev/peps/pep-0008/). You can enforce it easily with [`black`](https://black.readthedocs.io/en/stable/index.html) style and with [`flake8`](http://flake8.pycqa.org) conventions.
- [x] Please add tests to cover your changes, if applicable, and make sure that all new and existing tests pass locally.
- [x] If the behavior of the code has changed or new feature has been added, please also [update the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md).
- [x] Functions and classes have useful [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings and [type hints](https://www.python.org/dev/peps/pep-0484/) in the signature of the objects.
- [x] (Bug fix) The associated issue is referenced above using [auto-close keywords](https://help.github.com/en/articles/closing-issues-using-keywords). If the PR fixes an issue, use the keyword fix/fixes/fixed followed by the issue id. For example, if the PR fixes issue 1184, type "Fixes #1184" (without quotes).
- [x] The [changelog](https://github.com/unitaryfund/mitiq/blob/master/CHANGELOG.md) is updated, including author and PR number (@username, gh-xxx).

### CODE REVIEW

**Comment by purva-thakre** [comment]
I think this also fixes #591

File: docs/source/guide/guide-executors.rst [hunk 1]
```diff
@@ -20,191 +20,7 @@ Cirq Executors
 ======================================
 
 This section includes noisy and noiseless simulator executor examples using
```
Comment (rmlarose) [inline docs/source/guide/guide-executors.rst:22]:
```suggestion
```

Comment (rmlarose) [reply docs/source/guide/guide-executors.rst:22]: (there's no examples in this section anymore)

File: docs/source/guide/guide-executors.rst [hunk 2]
```diff
@@ -20,191 +20,7 @@ Cirq Executors
...
-
-    qc = Circuit()
-    for _ in range(4):
-        qc += cirq.X(LineQubit(0))
-    qc += cirq.measure(LineQubit(0))
-    qc = qc.with_noise(depolarize(p=0.02))
-    ham = cirq.PauliString(cirq.ops.Z.on(LineQubit(0)))
-    noisy_output = execute(qc, ham, 0.01, 200)
-    assert 0.5 < noisy_output < 1.0
+Cirq. For more, see module :mod:`mitiq_cirq.cirq_utils` for examples of Cirq executors.
```
Comment (rmlarose) [inline docs/source/guide/guide-executors.rst:23]:
```suggestion
See :mod:`mitiq_cirq.cirq_utils` for several Cirq executor functions.
```

File: mitiq/mitiq_cirq/cirq_utils.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        Returns:
+            The expectation value of obs as a float.
+        """
+
+    # Do the sampling
+    psum = cirq.PauliSumCollector(circ, obs, samples_per_term=shots)
+    psum.collect(sampler=cirq.Simulator())
+
+    # Return the expectation value
+    # Here, estimated_energy is a method of PauliSumCollector
```
Comment (rmlarose) [inline mitiq/mitiq_cirq/cirq_utils.py:57]:
```suggestion
```

File: mitiq/mitiq_cirq/cirq_utils.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            noise: The depolarizing noise as a float, i.e. 0.001 is 0.1% noise.
+
+        Returns:
+            The expectation value of obs as a float.
+        """
+    circuit = circ.with_noise(cirq.depolarize(p=noise))
+    simulator = cirq.DensityMatrixSimulator()
+    rho = simulator.simulate(circuit).final_density_matrix
+    # final_density_matrix and simulate belong to same module as
+    # DensityMatrixSimulator
```
Comment (rmlarose) [inline mitiq/mitiq_cirq/cirq_utils.py:78]:
```suggestion
```

File: mitiq/mitiq_cirq/cirq_utils.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        Args:
+            circ: The input Cirq circuit.
+            obs: The observable to measure as a NumPy array.
+            noise: The depolarizing noise strength as a float (0.001 is 0.1%)
+            shots: The number of measurements.
+
+        Returns:
+            The expectation value of obs as a float.
+        """
+    # add the noise
```
Comment (rmlarose) [inline mitiq/mitiq_cirq/cirq_utils.py:97]:
```suggestion
    # Add noise
```

File: mitiq/mitiq_cirq/cirq_utils.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    noisy = circ.with_noise(cirq.depolarize(p=noise))
+
+    # Do the sampling
+    psum = cirq.PauliSumCollector(noisy, obs, samples_per_term=shots)
+    psum.collect(sampler=cirq.DensityMatrixSimulator())
+
+    # Return the expectation value
+    # Here, estimated_energy is a method of PauliSumCollector
```
Comment (rmlarose) [inline mitiq/mitiq_cirq/cirq_utils.py:105]:
```suggestion
```

File: mitiq/mitiq_cirq/tests/test_cirq_utils.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    observable1 = np.diag([1, 0, 0, 0])
+    observable_exp_value = execute(qc, obs=observable1)
+    assert 0.0 == observable_exp_value
+
+    # Test 2 using another observable
+    new_qc = cirq.Circuit()
+    new_qc += [
+        cirq.X(cirq.LineQubit(0)),
+        cirq.CNOT(cirq.LineQubit(0), cirq.LineQubit(1)),
+    ]
```
[the remark this answers is not in the document]
Comment (rmlarose) [reply mitiq/mitiq_cirq/tests/test_cirq_utils.py:48]: this isn't used?

File: mitiq/mitiq_cirq/tests/test_cirq_utils.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+def test_execute_with_shots_and_depolarizing_noise():
+    """Tests if executor function for Cirq returns a proper expectation
+    value when used for noisy depoalrizing simulation and a finite number of
+    samples (aka shots)."""
+    shots = 200
+
+    qc = cirq.Circuit()
+    for _ in range(4):
+        qc += cirq.X(cirq.LineQubit(0))
+    qc += cirq.measure(cirq.LineQubit(0))
```
Comment (rmlarose) [inline mitiq/mitiq_cirq/tests/test_cirq_utils.py:106]: Could just be `qc = cirq.Circuit([cirq.X.on(q)] * 4, cirq.measure(q))` where `q = cirq.LineQubit(0)`.

File: mitiq/mitiq_cirq/tests/test_cirq_utils.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        qc, observable, noise2
+    )
+    assert np.isclose(observable_exp_value, 0.5)
+
+    # Test 3
+    noise3 = 0.001
+    observable_exp_value = execute_with_depolarizing_noise(
+        qc, observable, noise3
+    )
+    assert np.isclose(observable_exp_value, 0.062452)
```
Comment (rmlarose) [inline mitiq/mitiq_cirq/tests/test_cirq_utils.py:94]: I'm not sure the best way to test noisy simulation. These don't seem like the greatest tests to me, but I don't have any good recommendations at the moment.

File: mitiq/mitiq_cirq/tests/test_cirq_utils.py [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+    shots = 1000
+    observable = cirq.PauliString(
+        cirq.ops.Z.on(cirq.LineQubit(0)), cirq.ops.Z.on(cirq.LineQubit(1))
+    )
+    qc = cirq.Circuit()
+    qc += [
+        cirq.X(cirq.LineQubit(0)),
+        cirq.CNOT(cirq.LineQubit(0), cirq.LineQubit(1)),
+    ]
```
Comment (rmlarose) [inline mitiq/mitiq_cirq/tests/test_cirq_utils.py:61]: Also can just put in the constructor. `qc = cirq.Circuit(ops)`.

**Comment by purva-thakre** [comment]
> minor comments can be addressed in the future if desired

I could work on these later in the future or a separate issue could be created for someone else to tackle it ASAP. Currently I have enough issues to work on 😅 .

**Comment by crazy4pi314** [comment]
@all-contributors please add @purva-thakre for doc, infra, code, ideas

**Pull request merged**